File: src/88.合并两个有序数组.js
/*
 * @lc app=leetcode.cn id=88 lang=javascript
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let a = m-1, b = n-1
    while (b>=0 && a>=0) {
        if (nums1[a] >= nums2[b]) {
            nums1[a+b+1] = nums1[a]
            a--
        }
        else {
            nums1[a+b+1] = nums2[b]
            b--
        }
    }
    while (b>=0) {
        nums1[a+b+1] = nums2[b]
        b--
    }
};
// @lc code=end

File: src/88.合并两个有序数组.rs
/*
 * @lc app=leetcode.cn id=88 lang=rust
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let mut a = m as usize;
        let mut b = n as usize;

        while a>0 && b >0 {
            if nums1[a-1] > nums2[b-1] {
                nums1[a+b-1] = nums1[a-1];
                a-=1;
            } else {
                // println!("{}", b);
                nums1[a+b-1] = nums2[b-1];
                b-=1;
            }
        }

        while b>0 {
            nums1[b-1] = nums2[b-1];
            b-=1;
        }
    }
}
// @lc code=end

